packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -106,28 +105,42 @@ class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
     accessibility: treeItemBehavior,
   }
 
-  titleRef = React.createRef<HTMLElement>()
+  itemRef = React.createRef<HTMLElement>()
   treeRef = React.createRef<HTMLElement>()
 
   actionHandlers = {
-    getFocusFromParent: e => {
-      const { open } = this.props
-      if (open) {
+    performClick: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      _.invoke(this.props, 'onTitleClick', e, this.props)
+    },
+    collapseOrReceiveFocus: e => {
+      const { items, open } = this.props
+
+      e.preventDefault()
+
+      if (e.currentTarget !== e.target && items && items.length) {
```

Comment:
can you please add a comment to explain this condition, so it would be easier to get through in future

Reply:
sure, will add one.